rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -12,9 +12,175 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+find_package(rosidl_generator_c REQUIRED)
+find_package(rmw_implementation_cmake REQUIRED)
+find_package(rmw REQUIRED)
+
+# NOTE(esteve): required for CMake-2.8 in Ubuntu 14.04
+set(Python_ADDITIONAL_VERSIONS 3.4)
+find_package(PythonInterp 3.4 REQUIRED)
+
+find_program(PYTHON_CONFIG_EXECUTABLE NAMES "python3-config")
+if(NOT DEFINED PYTHON_CONFIG_EXECUTABLE)
+  message(FATAL_ERROR "Cannot find python3-config executable")
+endif()
+
+if(APPLE)
+  if(NOT DEFINED PYTHON_INCLUDE_DIRS)
+    execute_process(
+      COMMAND
+      "${PYTHON_CONFIG_EXECUTABLE}"
+      "--includes"
+      OUTPUT_VARIABLE _output
+      RESULT_VARIABLE _result
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    if(NOT _result EQUAL 0)
+      message(FATAL_ERROR
+        "execute_process(${PYTHON_CONFIG_EXECUTABLE} --includes returned "
+        "error code ${_result}")
+    endif()
+
+    string(REPLACE " " ";" _output_list ${_output})
+    set(PYTHON_INCLUDE_DIRS
+      ""
+      CACHE INTERNAL
+      "The paths to the Python include directories.")
+
+    foreach(_includedir ${_output_list})
+      string(SUBSTRING "${_includedir}" 2 -1 _includedir)
+      list(APPEND PYTHON_INCLUDE_DIRS "${_includedir}")
+    endforeach()
+  endif()
+
+  if(NOT DEFINED PYTHON_LIBRARIES)
+    execute_process(
+      COMMAND
+      "${PYTHON_CONFIG_EXECUTABLE}"
+      "--ldflags"
+      OUTPUT_VARIABLE _output
+      RESULT_VARIABLE _result
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    if(NOT _result EQUAL 0)
+      message(FATAL_ERROR
+        "execute_process(${PYTHON_CONFIG_EXECUTABLE} --libs returned "
+        "error code ${_result}")
+    endif()
+
+    string(REPLACE " " ";" _output_list ${_output})
+    set(PYTHON_LIBRARIES
+      ""
+      CACHE INTERNAL
+      "The libraries that need to be linked against for Python extensions.")
+
+    set(_library_paths "")
+    foreach(_item ${_output_list})
+      string(REGEX MATCH "-L(.*)" _regex_match ${_item})
+      if(NOT "${_regex_match} " STREQUAL " ")
+        string(SUBSTRING "${_regex_match}" 2 -1 _library_path)
+        list(APPEND _library_paths "${_library_path}")
+      endif()
+    endforeach()
+
+    set(_python_version_no_dots
+      "${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+    set(_python_version
+      "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
+
+    find_library(PYTHON_LIBRARY
+      NAMES
+      python${_python_version_no_dots}
+      python${_python_version}mu
+      python${_python_version}m
+      python${_python_version}u
+      python${_python_version}
+      PATHS
+      ${_library_paths}
+      NO_SYSTEM_ENVIRONMENT_PATH
+    )
+  endif()
+
+  set(PYTHON_LIBRARIES "${PYTHON_LIBRARY}")
```

Comment:
This block should print a similar message as the non-OSX branch and also state where the information was coming from.